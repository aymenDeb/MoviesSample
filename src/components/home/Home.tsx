import React, {Component} from 'react';
import {Text, SafeAreaView} from 'react-native';
import Header from '../commons/Header';
import SplashScreen from 'react-native-splash-screen';
import {getBestFilms} from '../../api/Api';
import FilmList from '../films/FilmList';
import Spinner from '../commons/Spinner';
import {connect} from 'react-redux';
import {getFavoriteList, setNetworkStatus} from '../../actions';
import {alertConnexion} from '../../constants/commonFunctions';
import styles from './HomeStyles';

interface Props {
  navigation: any;
  getFavoriteList: () => void;
  setNetworkStatus: () => void;
  isConnected: boolean;
}

interface State {
  films: any[];
  isLoading: boolean;
}

class Home extends Component<Props, State> {
  private totalPages: number;
  private page: number;
  constructor(props: Props) {
    super(props);
    this.page = 0;
    this.totalPages = 0;
    this.state = {
      films: [],
      isLoading: false,
    };
  }

  render(): JSX.Element {
    const {films, isLoading} = this.state;
    const {navigation, isConnected} = this.props;
    return (
      <SafeAreaView style={styles.container}>
        <Header />
        <Text style={styles.title}>Tendance actuelles {isConnected}</Text>
        <FilmList
          films={films}
          navigation={navigation}
          loadFilms={this.loadFilms}
          page={this.page}
          totalPages={this.totalPages}
        />
        {isLoading && isConnected ? <Spinner /> : null}
        {!isConnected ? alertConnexion(this.loadFilms) : null}
      </SafeAreaView>
    );
  }
  componentDidMount(): void {
    this.props.getFavoriteList();
    this.loadFilms();
    SplashScreen.hide();
  }

  loadFilms = () => {
    this.setState({isLoading: true});
    getBestFilms(this.page + 1, this.props.setNetworkStatus).then((data) => {
      this.page = data.page;
      this.totalPages = data.total_pages;
      this.setState({
        films: [...this.state.films, ...data.results],
        isLoading: false,
      });
    });
  };
}

const mapStateToProps = (state: any) => {
  return {
    isConnected: state.network.isConnected,
  };
};

export default connect(mapStateToProps, {getFavoriteList, setNetworkStatus})(
  Home,
);
