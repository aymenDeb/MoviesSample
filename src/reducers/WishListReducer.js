import {ADD_FILM_LIST, GET_LIST_FILM} from '../actions/types';

const INITIAL_STATE = {favoritesFilm: []};

export default (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case ADD_FILM_LIST: {
      return {
        ...state,
        favoritesFilm: action.payload,
      };
    }
    case GET_LIST_FILM: {
      return {
        ...state,
        favoritesFilm: action.payload,
      };
    }

    default:
      return state;
  }
};
