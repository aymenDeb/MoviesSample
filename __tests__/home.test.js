import React from 'react';
import Home from '../src/components/home/Home';
import {shallow} from 'enzyme';
import WishListReducer from '../src/reducers/WishListReducer';
import ConnectComponent from '../__mocks__/connectComponent';

describe('Test sur Home', () => {
  it('render composant connectÃ©', () => {
    const component = shallow(
      <ConnectComponent>
        <Home />
      </ConnectComponent>,
    );
    expect(component).toMatchSnapshot();
  });
  it('Test le reducer sans action type', () => {
    const initial_state = {favoritesFilm: []};
    expect(WishListReducer(initial_state, {}).favoritesFilm).toEqual([]);
  });
});
