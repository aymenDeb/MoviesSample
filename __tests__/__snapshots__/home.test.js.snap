// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test sur Home render composant connect√© 1`] = `ShallowWrapper {}`;
